concent_api/core/tests/test_utils.py

```diff
@@ -0,0 +1,107 @@
+
+        mock_maximum_download_time.assert_called_once_with(
+            1,
+            settings.MINIMUM_UPLOAD_RATE,
+        )
+
+    @override_settings(
+        CUSTOM_PROTOCOL_TIMES=True
+    )  # pylint: disable=no-self-use
+    def test_that_calculate_maximum_download_time_will_call_maximum_download_time_when_custom_protocol_times_is_on(self):
+        with mock.patch('core.utils.maximum_download_time', return_value=datetime.timedelta()) as mock_maximum_download_time:
+            calculate_maximum_download_time(size=1)
+
+        mock_maximum_download_time.assert_not_called()
+
+    @override_settings(
+        DOWNLOAD_LEADIN_TIME=int(constants.DOWNLOAD_LEADIN_TIME.total_seconds())
+    )
+    def test_that_both_maximum_download_time_implementation_should_return_same_result_when_golem_messages_constants_match_concent_settings(self):
+        for size, rate in [
+            (10, 10),
+            (100, 100),
+            (1000, 1000),
+            (9999, 9999),
+        ]:
+            self.assertEqual(
+                calculate_maximum_download_time(size, rate),
+                int(helpers.maximum_download_time(size, rate).total_seconds())
+            )
+            self.assertEqual(
+                calculate_maximum_download_time(size),
+                calculate_maximum_download_time(size, constants.DEFAULT_UPLOAD_RATE),
+            )
+
+
+class CalculateSubtaskVerificationTimeTestCase(ConcentIntegrationTestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.report_computed_task = self._get_deserialized_report_computed_task()
+
+    @override_settings(
+        CUSTOM_PROTOCOL_TIMES=False
+    )
+    def test_that_calculate_subtask_verification_time_will_call_subtask_verification_time_when_custom_protocol_times_is_off(self):
+        with mock.patch('core.utils.subtask_verification_time', return_value=datetime.timedelta()) as mock_subtask_verification_time:
+            calculate_subtask_verification_time(self.report_computed_task)
+
+        mock_subtask_verification_time.assert_called_once_with(self.report_computed_task)
+
+    @override_settings(
+        CUSTOM_PROTOCOL_TIMES=True
+    )
+    def test_that_calculate_subtask_verification_time_will_call_subtask_verification_time_when_custom_protocol_times_is_on(self):
+        with mock.patch('core.utils.subtask_verification_time', return_value=datetime.timedelta()) as mock_subtask_verification_time:
+            calculate_subtask_verification_time(self.report_computed_task)
+
+        mock_subtask_verification_time.assert_not_called()
+
+    @override_settings(
+        CONCENT_MESSAGING_TIME=int(constants.CMT.total_seconds())
+    )
+    def test_that_both_subtask_verification_time_implementation_should_return_same_result_when_golem_messages_constants_match_concent_settings(self):
+        for size, timestamp, deadline in [
+            (10,    get_current_utc_timestamp(),            get_current_utc_timestamp() + 10),
```

Comment:
please extract `get_current_utc_timestamp()` to a variable and reuse it

Reply:
:+1: